DQMOffline/Trigger/src/HLTMuonMatchAndPlot.cc

```diff
@@ -273,12 +279,34 @@ void HLTMuonMatchAndPlot::analyze(Handle<MuonCollection>& allMuons,
     nonDZPath = nonDZPath.substr(0, nonDZPath.rfind("_v") + 2);
   }
   bool passTriggerDZ = false;
+
   if (dzPath) {
     for (unsigned int hltIndex = 0; hltIndex < numTriggers; ++hltIndex) {
       passTriggerDZ = passTriggerDZ || (trigNames.triggerName(hltIndex).find(nonDZPath) != std::string::npos &&
                                         triggerResults->wasrun(hltIndex) && triggerResults->accept(hltIndex));
     }
   }
+  if (dzPath && targetMuons.size() > 1 && passTriggerDZ) {
+    const Track* track0 = nullptr;
+    const Track* track1 = nullptr;
+    if (targetMuons.at(0).isTrackerMuon())
+      track0 = &*targetMuons.at(0).innerTrack();
+    else if (targetMuons.at(0).isStandAloneMuon())
+      track0 = &*targetMuons.at(0).outerTrack();
+    if (targetMuons.at(1).isTrackerMuon())
+      track1 = &*targetMuons.at(1).innerTrack();
+    else if (targetMuons.at(1).isStandAloneMuon())
+      track1 = &*targetMuons.at(1).outerTrack();
```

Comment:
```suggestion
    auto track_ptr = [](reco::Muon const& muo){
      if (muo.isTrackerMuon())
        return &*muo.innerTrack();
      else if (muo.isStandAloneMuon())
        return &*muo.outerTrack();
      return nullptr;
    }
    auto const* track0 = track_ptr(targetMuons[0]);
    auto const* track1 = track_ptr(targetMuons[1]);
```
Arguably more readable, I have not tested it.